qutebrowser/browser/adblock.py

```diff
@@ -61,15 +60,18 @@ def get_fileobj(byte_io):
     return byte_io
 
 
-def is_whitelisted_host(host):
-    """Check if the given host is on the adblock whitelist.
+def is_whitelisted_host(url):
+    """Check if the given url is on the adblock whitelist.
 
     Args:
-        host: The host of the request as string.
+        url: The url to check in QUrl form.
     """
     for pattern in config.val.content.host_blocking.whitelist:
-        if fnmatch.fnmatch(host, pattern.lower()):
-            return True
+        try:
+            if pattern.matches(url):
+                return True
+        except urlmatch.ParseError:
```

Comment:
This should never happen, as with the `UrlPattern` config type, the config system will make sure that you don't get any invalid patterns. I'll remove it.